ggml/src/CMakeLists.txt

```diff
@@ -504,7 +506,8 @@ if (GGML_HIPBLAS)
         message(FATAL_ERROR "Static linking not supported for HIP/ROCm")
     endif()
 
-    set(GGML_EXTRA_LIBS ${GGML_EXTRA_LIBS} PUBLIC hip::host roc::rocblas roc::hipblas)
+    # TODO: this "PUBLIC" here seems wrong
+    list(APPEND GGML_EXTRA_LIBS PUBLIC hip::host roc::rocblas roc::hipblas)
```

Comment:
I am surprised this hack works at all. But I guess its strings internally...
But that also means that everything following will be made public.

Reply:
Yup me too. I don't have HIP environment setup to test and find ways to avoid it.